### DISCUSSION TITLE
region endpoint available?

### DISCUSSION THREAD

**Discussion by Delta0S** [isFakeAccount/psnawp#83] [Q&A]
Hi,

I am using your PSN API wrapper and I need to access the region of a user endpoint. I looked through the documentation and code but couldn't find it. Could you please let me know if this endpoint is available or if there is a way to retrieve the user's region through your API?

**Comment by isFakeAccount** [comment]
@Delta0S Can you show me on mobile app how to get region or whatever you want. 

Profile endpoint can return language. Idk if that will be enough for you. 

https://psnawp.readthedocs.io/en/latest/psnawp_api.models.html#psnawp_api.models.user.User.profile

**Reply by Delta0S** [reply]
> Great I'll check it out. I'm still learning, but one of the checker owners advised me to 'decode npId using base64 decoding.' You can find npId in the API documentation at https://psn-api.achievements.app/api-docs/users.

![IMG_20240719_153904_978](https://github.com/user-attachments/assets/ff7cc2b3-1b40-4fda-a44a-f73d44fd4e23)

**Reply by isFakeAccount** [reply]
> > Great I'll check it out. I'm still learning, but one of the checker owners advised me to 'decode npId using base64 decoding.' You can find npId in the API documentation at https://psn-api.achievements.app/api-docs/users.
> 
> ![IMG_20240719_153904_978](https://github.com/user-attachments/assets/ff7cc2b3-1b40-4fda-a44a-f73d44fd4e23)
> 

So this function is available in Client class. It is called `get_profile_legacy()`. You can port it to User class and make a PR. @Delta0S 

Read the contributing.md. If you have any questions let me know.

**Reply by Delta0S** [reply]
hi, I'm not sure how to move the function to the `User` class. I use AI to help me with coding and I'm still a beginner. You can check my work here: [PSN Checker Discord](https://github.com/Delta0S/PSN-Checker-Discord/tree/main). Any guidance would be appreciated.

**Reply by isFakeAccount** [reply]
> hi, I'm not sure how to move the function to the `User` class. I use AI to help me with coding and I'm still a beginner. You can check my work here: [PSN Checker Discord](https://github.com/Delta0S/PSN-Checker-Discord/tree/main). Any guidance would be appreciated.

@Delta0S I didn't say move. I said copy it from Client class to User class.
